projects/jobs/jobs.groovy

```diff
@@ -23,86 +33,174 @@ loadCartridgeJob.with{
+      }
+    }
+  }
+  catch (UnknownHostException e) {
+    cartridge_urls.add("[ERROR] Provided URL was not found: ${source_url}");
+    println "[ERROR] Provided URL was not found: ${source_url}";
+  }
+  catch (Exception e) {
+    cartridge_urls.add("[ERROR] Unknown error while processing: ${source_url}");
+    println "[ERROR] Unknown error while processing: ${source_url}";
+  }
+}
 
-              return cartridge_urls;
+return cartridge_urls;
 ''')
               defaultChoice('Top')
               usePredefinedVariables(false)
             }
           }
           editable(true)
           description('Cartridge URL to load')
-          }
+        }
+        activeChoiceParam('SCM_PROVIDER') {
+          description('Your chosen SCM Provider and the appropriate cloning protocol')
+          filterable()
+          choiceType('SINGLE_SELECT')
+          scriptlerScript('retrieve_scm_props.groovy')
+        }
+        if (customScmNamespace == "true"){
+          stringParam('SCM_NAMESPACE', '', 'The namespace for your SCM provider which will prefix your created repositories')
+        }
         stringParam('CARTRIDGE_FOLDER', '', 'The folder within the project namespace where your cartridge will be loaded into.')
         stringParam('FOLDER_DISPLAY_NAME', '', 'Display name of the folder where the cartridge is loaded.')
         stringParam('FOLDER_DESCRIPTION', '', 'Description of the folder where the cartridge is loaded.')
         booleanParam('ENABLE_CODE_REVIEW', false, 'Enables Gerrit Code Reviewing for the selected cartridge')
         booleanParam('OVERWRITE_REPOS', false, 'If ticked, existing code repositories (previously loaded by the cartridge) will be overwritten. For first time cartridge runs, this property is redundant and will perform the same behavior regardless.')
     }
     environmentVariables {
+        groovy("return [SCM_KEY: org.apache.commons.lang.RandomStringUtils.randomAlphanumeric(20)]")
         env('WORKSPACE_NAME',workspaceFolderName)
         env('PROJECT_NAME',projectFolderName)
     }
     wrappers {
         preBuildCleanup()
         injectPasswords()
         maskPasswords()
-        sshAgent("adop-jenkins-master")
+        credentialsBinding {
+            file('SCM_SSH_KEY', 'adop-jenkins-private')
+        }        
+        copyToSlaveBuildWrapper {
+          includes("**/**")
+          excludes("")
+          flatten(false)
+          includeAntExcludes(false)
+          relativeTo('''${JENKINS_HOME}/userContent''')
+          hudsonHomeRelative(false)
+        }
     }
+    label("!master")
     steps {
         shell('''#!/bin/bash -ex
 
+mkdir ${WORKSPACE}/tmp
+
+# Output SCM provider ID to a properties file
+echo SCM_PROVIDER_ID=$(echo ${SCM_PROVIDER} | cut -d "(" -f2 | cut -d ")" -f1) > scm.properties
+
+# Check if SCM namespace is specified
+if [ -z ${SCM_NAMESPACE} ] ; then
+    echo "SCM_NAMESPACE not specified, setting to PROJECT_NAME..."
+    SCM_NAMESPACE="${PROJECT_NAME}"
```

Comment:
SCM_NAMESPACE="${PROJECT_NAME}/${CARTRIDGE_FOLDER}" if cartridge folder has been specified. If not I suspect it will just be empty and default to using the project name.